node/src/components/consensus/era_supervisor.rs

```diff
@@ -370,6 +373,22 @@ where
     pub(crate) fn stop_for_upgrade(&self) -> bool {
         self.stop_for_upgrade
     }
+
+    fn executed_block(&mut self, block_header: &BlockHeader) {
+        self.next_executed_height = self.next_executed_height.max(block_header.height() + 1);
```

Comment:
Is it possible for the `block_header.height()` to be lower than `self.next_executed_height`?

Reply:
I think it might happen, during joining.
But whenever you're an active, fully synced, participating validator that is in a position to actually contribute to consensus, the latest executed height will be slightly behind the latest finalized one.